src/arrayMethodJoin.js

```diff
@@ -5,7 +5,18 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    const result = [...this].toString();
+
+    switch (separator) {
+      case null:
+        return result.replaceAll(',', 'null');
```

Comment:
do not use other methods for implementing custom method.
just base js tools (`for loop`, `if`...)